sdks/python/apache_beam/runners/common.py

```diff
@@ -95,17 +95,20 @@ def __init__(self, do_fn):
     self._validate()
 
   def _validate(self):
+    self._validate_process()
     self._validate_bundle_method(self.start_bundle_method)
     self._validate_bundle_method(self.finish_bundle_method)
 
-  def _validate_bundle_method(self, method_wrapper):
-    # Here we use the fact that every DoFn parameter defined in core.DoFn has
-    # the value that is the same as the name of the parameter and ends with
-    # string 'Param'.
-    unsupported_dofn_params = [i for i in core.DoFn.__dict__ if
-                               i.endswith('Param')]
+  def _validate_process(self):
+    """Validate that none of the DoFnParameters are repeated in the function
+    """
+    for param in core.DoFn.DoFnParams:
```

Comment:
I don't have strong opinions but I feel if the end user is asking for the same value twice that just seems wrong if they are hoping to get different window values or something. 

How about we keep this as making it more lenient is easy?